src/util.test.ts

```diff
@@ -1068,23 +1073,129 @@ describe('util', () => {
+        util.getFormattedIpfsUrl(
+          `https://${IFPS_GATEWAY}`,
+          `${DEFAULT_IPFS_URL_FORMAT}${IPFS_CID_V1}/test`,
+          true,
+        ),
+      ).toStrictEqual(`https://${IPFS_CID_V1}.ipfs.${IFPS_GATEWAY}/test`);
+    });
+
+    it('should return a correctly formatted non-subdomained ipfs url when passed ipfsGateway with no "/ipfs/" appended, a path at the end of the url, and subdomainSupported argument set to false', () => {
+      expect(
+        util.getFormattedIpfsUrl(
+          `https://${IFPS_GATEWAY}`,
+          `${DEFAULT_IPFS_URL_FORMAT}${IPFS_CID_V1}/test`,
+          false,
+        ),
+      ).toStrictEqual(`https://${IFPS_GATEWAY}/ipfs/${IPFS_CID_V1}/test`);
     });
 
-    it('should return content identifier from alternative ipfs url', () => {
+    it('should return a correctly formatted non-subdomained ipfs url when passed an ipfsGateway with "/ipfs/" appended, a path at the end of the url, subdomainSupported argument set to false', () => {
       expect(
-        util.getIpfsUrlContentIdentifier(ALTERNATIVE_IPFS_URL),
-      ).toStrictEqual('0001');
+        util.getFormattedIpfsUrl(
+          `https://${IFPS_GATEWAY}/ipfs/`,
+          `${DEFAULT_IPFS_URL_FORMAT}${IPFS_CID_V1}/test`,
+          false,
+        ),
+      ).toStrictEqual(`https://${IFPS_GATEWAY}/ipfs/${IPFS_CID_V1}/test`);
     });
+  });
 
-    it('should return url if its not a ipfs standard url', () => {
-      expect(util.getIpfsUrlContentIdentifier(SOME_API)).toStrictEqual(
-        SOME_API,
+  describe('removeIpfsProtocolPrefix', () => {
+    it('should return content identifier and path combined string from default ipfs url format', () => {
+      expect(
+        util.removeIpfsProtocolPrefix(
+          `${DEFAULT_IPFS_URL_FORMAT}${IPFS_CID_V0}/test`,
+        ),
+      ).toStrictEqual(`${IPFS_CID_V0}/test`);
+    });
+
+    it('should return content identifier string from default ipfs url format if no path preset', () => {
+      expect(
+        util.removeIpfsProtocolPrefix(
+          `${DEFAULT_IPFS_URL_FORMAT}${IPFS_CID_V0}`,
+        ),
+      ).toStrictEqual(IPFS_CID_V0);
+    });
+
+    it('should return content identifier string from alternate ipfs url format', () => {
+      expect(
+        util.removeIpfsProtocolPrefix(
+          `${ALTERNATIVE_IPFS_URL_FORMAT}${IPFS_CID_V0}`,
+        ),
+      ).toStrictEqual(IPFS_CID_V0);
+    });
+
+    it('should throw error if passed a non ipfs url', () => {
+      expect(() => util.removeIpfsProtocolPrefix(SOME_API)).toThrow(
+        'this method should not be used with non ipfs urls',
+      );
+    });
+  });
+
+  describe('addUrlProtocolPrefix', () => {
+    it('should return a URL with https:// prepended if input URL does not already have it', () => {
+      expect(util.addUrlProtocolPrefix(IFPS_GATEWAY)).toStrictEqual(
+        `https://${IFPS_GATEWAY}`,
       );
     });
+
+    it('should return a URL as is if https:// is already prepended', () => {
```

Comment:
Nit: It's probably also worth ensuring that `http` URLs are handled correctly as well

Reply:
Will handle in follow up PR!